### DISCUSSION TITLE
FetchImage usage

### DISCUSSION THREAD

**Discussion by jose-pay** [kean/Nuke#673] [General]
I need an image to load a alternative URL if the first one fails. It should also show a placeholder image if both URLs fail. I am using `FetchImage` for this:

    struct ProfileImageView: View {

        @StateObject private var image = FetchImage()

        let imageURL: URL
        let alternativeImageURL: URL

        var body: some View {
            ZStack {
                image.view?
                    .onFailure { _ in

                    }
                    .animation(.easeInOut)
                    .frame(width: 40, height: 40)
                    .clipShape(Circle())
            }
            .onAppear {
                image.load(imageURL)
            }
        }
    }

Questions 

1. I need to wrap the `image.view?` inside some other view (`ZStack` in this case) to use `onAppear`, correct? 
2. I am not sure where I can add the `onFailure` modifier to load the alternative image 
3. Where can I put/define the placeholder `Image` view?

Thanks.

**Comment by kean** [comment]
I suggest using `LazyImage` from the [nuke-12](https://github.com/kean/Nuke/tree/nuke-12) branch as a reference. It has a couple of optimizations you don't necessarily need, e.g. memory cache lookup on the first `body` call. You can safely skip these.

1. That's why `LazyImage` does; works OK
2. You can do it either in `onAppear` if you are using `@StateObject` or when creating `FetchImage` if you are injecting it using `@ObservedObject`
3. See https://github.com/kean/Nuke/blob/f0b411f6770b81ad9d3ec35292e5ff16beb92d7c/Sources/NukeUI/LazyImage.swift#L146. You might also want to add an error state based on the `FetchImage` state
